langchain-python/run.sh

```diff
@@ -7,21 +7,18 @@ set -x
 
 PYTHON_BINARY=$(find_python3)
 
-cd libs/langchain
+cd libs/partners/mongodb
 
 $PYTHON_BINARY -m venv .
 source ./bin/activate
 
 pip install poetry
-pip install filelock
-pip install msgpack
-pip install requests_toolbelt
-pip install motor
 
 poetry install --with test
 # Workaround provided in https://github.com/langchain-ai/langchain/issues/12237
```

Comment:
Should we remove this block now that we don't need a workaround?

Reply:
According to the Poetry docs, "...the isolated virtual environment in which poetry is installed should not be activated for running poetry commands." 

I'm going through this stuff now. What they recommend is more like this

```
PYTHON_BINARY -m venv $VENV_PATH
$VENV_PATH/bin/pip install -U pip setuptools
$VENV_PATH/bin/pip install poetry 
$VENV_PATH/bin/pip install poetry install --with test
```

Reply:
The following appears to work well for me.

```
PYTHON_BINARY=$(find_python3)
$PYTHON_BINARY -c "import sys; print(f'python version found: {sys.version_info}')"

$PYTHON_BINARY -m pip install poetry
poetry env use python3.10
source $(poetry env info --path)/bin/activate
poetry install --with dev

# Stored in evergreen VARIABLES
# source sandbox/setenv.sh. <== Disregard. This was from running locally. I'm not set up with CI approach just yet.

poetry run pytest tests/datastore/providers/mongodb_atlas/
```

Reply:
How do we actually set the python version?

Reply:
The venv python version? In my version, I'm still keeping to the poetry heuristic. I'm just walling everything off with a venv.

Reply:
Hey. This was an OLD comment, I think. The code in `find_python3` is pretty simple.